### DISCUSSION TITLE
install ati driver for thin client

### DISCUSSION THREAD

**Discussion by backtolinux** [ltsp/ltsp#290] [Q&A]
Hello everybody,

first, thanks for author for the product.
Second, thanks to people involved and participating.
last, I succesfully installed, configured and run server and thin client.
My hardware are far far away from what indicated in https://ltsp.org/advanced/specs:
My server is AMD phenom X4 9550 with 16Go ram. (less than thinclient indicated lol)
Running Debian 10 under KVM (with 2 Go ram) bridged.
The thinclient is Wyse R90L AMD Sempron 200U (monocore 64bits), 4Go RAM and CF PATA 16Go and graphic card is RS690 (X1xxx).
It worked pretty well but I wanted thinclient play youtube.
To have correct performance, I wanted to install ati proprietary driver from https://www.amd.com/fr/support/graphics/legacy-graphics/ati-radeon-x-series/ati-radeon-x1xxx-series.
I don't know how to install it.
Is a possibility to do it ?
Mounting .img in loop ?
thanks in advance for the help.
Kind regards.

**Comment by alkisg** [comment]
Hi, if you're using the chrootless setup, install them on the server and run `ltsp image /`;
if you're using a VM, install them in the vm and run `ltsp image vm-name`.

**Comment by backtolinux** [comment]
Hi, thanks for your answer.
But, I'm not sure it can be fit :
the PC is under ubuntu 18.04 in which i installed KVM.
I created a VM with debian 10 in which I installed ltsp server.
The VM created has qcow2 file and I don't want to change format to vmblabla (not solo reason : my project is to run ltsp inside container lxc/lxd and have HA for cloud or HA i.e.).
In this VM (QXL graphic cards), I'm not sure if I can install proprietary driver which come with firmware to load at startup.
I'm pretty sure system will deny the installation of these driver (how to configure them after all because will be lost in client side) or will failed to boot (at least display manager).
Drivers + firmware on NFS, It will pain full to install at each reboot. by number of client, more paint full.
A possibility to mount rw the image and chroot inside would be nice.
I tested the mount with rw option, but the chrooted file system still remains ro.

**Comment by alkisg** [comment]
AFAIK nvidia and amd drivers have been able to co-exist with xorg drivers without breaking things for a few years now. I.e. it should be possible to install nvidia and amd drivers in the same image, and then boot an intel client from it, without issues.

To chroot to your .qcow2 file, you'd boot the clients, access the .qcow2 over SSHFS or NFS, and run [guestfish](http://libguestfs.org/guestfish.1.html) or any other tool that can mount .qcow2 images. It would be a lot easier if you used a chroot or a -flat.vmdk file.

The /srv/ltsp/images/image.img images are squashfs, which is read-only, not modifiable; you want to modify the .qcow2 image (or chroot or vmdk), not the squashfs one.

**Comment by backtolinux** [comment]
Not sure to understand ?
For my point of view:
The VM used QXL/Splice driver which are incompatible with proprietary driver nor open driver.
I don't have passthrough features in this PC, no vt-d.
None nvidia nor ati involved in the qemu-KVM graphic driver.

Image customization could be good for thinclient in different model.
My model WYSE R90L is without sata, another model exists with it.
Only 2 numbers changed without noticed by editor but still same reference.

I'm not close and I'll try to install ati driver.
modifying qcow2 to vmdk if it doesn't work and come back with result.

**Comment by alkisg** [comment]
> The VM used QXL/Splice driver which are incompatible with proprietary driver nor open driver.

What is the error message when you try to install both of them?
Normally, it's possible to have 10 xorg drivers installed. Xorg automatically loads the correct one.

> Image customization could be good for thinclient in different model.

I don't understand what you're asking. To be able to modify the squashfs images instead of the VM images?

Normally, we either have many images,
or we put everything in one image (all drivers, all programs etc), and each client uses the one it needs.

**Comment by backtolinux** [comment]
Hello,
i checked VM and firmware-amd-graphics already installed.
My real problem was that I cannot open browser without overloading CPU ( simple core to 80-90% used in top)
That why I searched a solution.
And ltsp (nor another soft I tried) didn't resolve my problem. Even installing driver and firmware.
Always decoding video occurred on thin-client.
I made a try to install natively debian on thinclient, and problem still here : webcontent "eat" completely the CPU. Even without youtube or netflix.
PC became impracticable with one of these 2 video site.
I finally resolved my issue by using tigervnc and its capacity to open desktop others than local display :0 and per user.
And, all are really managed on server. Only showing desktop on thin client.
Only my though and for me and my point of view, that was what I wanted from archi server/thinclient
Sorry for inconvenient.
thanks at all.

**Comment by alkisg** [comment]
LTSP is about netbooting clients.
If after you boot them, you want to do "remote desktop" to the server, you can use VNC, RDP, X2GO, X -query, ssh -X, or any other remote desktop software.
See [this discussion](https://github.com/ltsp/community/issues/32) for more.

**Comment by backtolinux** [comment]
Hi,
I've just wanted to play video-site in the thinclient.
each time browser are launching, webcontent and x-www-browser consum near all the cpu (between 70 to 90 of the whole).

What my understanding (maybe i'm wrong) for terminal server architecture with thin client is, the thin client display and no compute.
Thin client is a graphical console, not more, not less.

With ltsp, browser used huge cpu and memory (even with localy installed distr). This not suite to me.
For me that means, thin client is a "fat client".

Vnc, a contrario, let all program executed on server and the thin client only display result (in a window), that thing i need.

In case of ltsp server crash for whatever reason, thin client become useless.
That why, I installed localy a distrib (debian because i like, i tried puppy too) and I'm able to do many thing which not need too much power (mail, browse some forum, wiki, but never video-site).

Sure ltsp could be installed and in case server not available, thin client boot locally.
That why i will not uninstall ltsp from my VM. But, live CD did near the same.

All are my understanding and my point of view.

Each one has his own problem with a solution that fit to him, but not necessary suite for other.

Kind regards.

**Comment by alkisg** [comment]
Yes, LTSP is about netbooting fat clients, not thin clients.

You can use remote desktop programs like VNC, RDP, X2GO, X -query, ssh -X, to make them thin and have tasks run on the server.
